### PR TITLE
[receiver/hostmetrics] make sure process scraper uses root_path config entry

### LINKED ISSUE #24777: [receiver/hostmetrics] Process scraper ignores root_path when getting process information
### Component(s)

receiver/hostmetrics

### What happened?

I am using the `hostmetrics` process scraper running inside a Docker container on Linux. My receiver configuration is as follows:

```
receivers:
  hostmetrics: 
    collection_interval: 60s
    root_path: /hostfs
    scrapers:
      # cpu, disk, etc. omitted here
      process:
        mute_process_exe_error: false # 0.82 only
        mute_process_name_error: false
```

My docker container is set up in a docker-compose file as follows:

```
 otel-collector:
    image: ${OTEL-IMAGE-NAME-HERE}
    container_name: otel-collector
    mem_limit: 100m
    user: "0"
    memswap_limit: -1
    command: ["--config=/etc/otel-collector-config.yaml"]
    volumes:
      - ./agent/config/otel-collector.yaml:/etc/otel-collector-config.yaml
      - /proc:/hostfs/proc
```

Previously I was using version 0.71 which did not have the `mute_process_exe_error` parameter, so I did not have it in the config at the time. When attempting to scrape processes I would receive the following error message for every process running on my machine:

```
2023-08-02T00:37:21.762Z        error   scraperhelper/scrapercontroller.go:212  Error scraping metrics  {"kind": "receiver", "name": 
"hostmetrics", "data_type": "metrics", "error": "error reading process name for pid 1: readlink /hostfs/proc/1/exe: permission denied; 
error reading process name for pid 2: readlink /hostfs/proc/2/exe: permission denied; error reading process name for pid 3: readlink
 /hostfs/proc/3/exe: permission denied; error reading process name for pid 4: readlink /hostfs/proc/4/exe: permission denied;
```

These are expected errors that have since been rectified with the `mute_process_exe_error` parameter. In particular the process names are reading `/hostfs/proc/[pid]` which is the expected path since that is how I set up the bind mount.

After updating from version 0.71 to 0.82, keeping everything in the otel config and the docker-compose file the same (except adding the `mute_process_exe_error` which I have set to false while troubleshooting this issue), I see the following error instead:

```
2023-08-02T00:16:30.691Z        error   scraperhelper/scrapercontroller.go:200  Error scraping metrics  {"kind": "receiver", "name": 
"hostmetrics", "data_type": "metrics", "error": "error reading username for process \"otelcol-contrib\" (pid 1): open /etc/passwd: 
no such file or directory; error reading process executable for pid 2: readlink /proc/2/exe: no such file or directory; error reading 
process name for pid 2: open /proc/2/status: no such file or directory; error reading process executable for pid 3: readlink 
/proc/3/exe: no such file or directory; error reading process name for pid 3: open /proc/3/status: no such file or directory; error 
reading process executable for pid 4: readlink /proc/4/exe: no such file or directory; error reading process name for pid 4: open 
/proc/4/status: no such file or directory; error reading username for process \"otelcol-contrib\" (pid 6): open /etc/passwd: no 
such file or directory;
```

Of note, the path it is looking at now is `/proc/[pid]` rather than `hostfs/proc/[pid]`, and the processes it finds are internal container processes (since all of the process names are `otelcol-contrib`. For example pid 1 on my machine is the default Linux process /sbin/init). My guess is that the scraper is able to read all of the processes on the machine that have been mounted onto the docker container, but to get all of the process information it still goes to /proc (where the internal container processes are) instead of /hostfs/proc. If I mute the exe and name errors, the few metrics I get are for processes that are all labeled `otelcol-contrib` with low pid's, which also makes me think these are process metrics from within the container itself.

### Collector version

v0.71.0, 0.82.0

## Environment
OS: Ubuntu 18.04

### OpenTelemetry Collector configuration

```yaml
receivers:
  hostmetrics: 
    collection_interval: 60s
    root_path: /hostfs
    scrapers:
      # cpu, disk, etc. omitted here
      process:
        mute_process_exe_error: false # 0.82 only
        mute_process_name_error: false

processors:

exporters:
  logging:
    verbosity: normal
  prometheusremotewrite:
    # 

service:
  pipelines:
    metrics:
      receivers: [hostmetrics]
      processors: []
      exporters: [logging, prometheusremotewrite]
```

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
A regression introduced in 0.82.0 means that the process scraper doesn't properly respect the `root_path` configuration key when it comes to reading process information.

**Link to tracking Issue:**
Fixes #24777 

**Testing:**
Unit tests fixes and new end to end tests.

**Documentation:**
N/A

### CODE REVIEW

File: receiver/hostmetricsreceiver/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build e2e
+// +build e2e
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/e2e_test.go:5]: This won't be run on CI. We only call it for k8s components explicitly. Let's put `integration` tag here

**Review by dmitryax — COMMENTED** [review]
@atoulme please rebase

**Comment by dmitryax** [comment]
@atoulme please rebase

**Comment by atoulme** [comment]
@dmitryax ready for you I think?

**Review by mx-psi — APPROVED** [review]

File: receiver/hostmetricsreceiver/go.mod [hunk 2]
```diff
@@ -5,7 +5,9 @@ go 1.20
 require (
 	github.com/google/go-cmp v0.5.9
 	github.com/leoluk/perflib_exporter v0.2.1
-	github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter v0.85.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.85.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter v0.84.0
```
[the remark this answers is not in the document]
Comment (atoulme) [reply receiver/hostmetricsreceiver/go.mod]: Good catch, fixed :)

**Review by dmitryax — APPROVED** [review]

**Pull request merged**